src/message.rs

```diff
@@ -77,18 +77,54 @@ impl MsgId {
         Ok(result)
     }
 
-    /// Returns true if the message needs to be moved from `folder`.
-    pub async fn needs_move(self, context: &Context, folder: &str) -> Result<bool, Error> {
-        if !context.get_config_bool(Config::MvboxMove).await {
-            return Ok(false);
-        }
-
+    /// Returns Some if the message needs to be moved from `folder`.
+    /// If yes, returns `ConfiguredInboxFolder`, `ConfiguredMvboxFolder` or `ConfiguredSentboxFolder`,
+    /// depending on where the message should be moved
+    pub async fn needs_move(
+        self,
+        context: &Context,
+        folder: &str,
+    ) -> Result<Option<Config>, Error> {
+        use Config::*;
         if context.is_mvbox(folder).await {
-            return Ok(false);
+            return Ok(None);
         }
 
         let msg = Message::load_from_db(context, self).await?;
 
+        if context.is_spam_folder(folder).await {
+            return if msg.chat_blocked == Blocked::Not {
+                if self.needs_move_to_mvbox(context, &msg).await? {
+                    Ok(Some(ConfiguredMvboxFolder))
+                } else {
+                    Ok(Some(ConfiguredInboxFolder))
+                }
+            } else {
+                // Blocked/deaddrop message in the spam folder, leave it there
+                Ok(None)
+            };
+        }
+
+        if self.needs_move_to_mvbox(context, &msg).await? {
+            Ok(Some(ConfiguredMvboxFolder))
+        } else if msg.state.is_outgoing()
+                && msg.is_dc_message == MessengerMessage::Yes
+                && !msg.is_setupmessage()
+                && msg.to_id != DC_CONTACT_ID_SELF // Leave self-chat-messages in the inbox, not sure about this
```

Comment:
I would rather have BCC-self messages in DeltaChat folder if it is configured.

Reply:
If `mvbox_move` is on, we will already move both self-sent and BCC-self messages to the DeltaChat folder.

This line is about where to put self-chat "Saved messages" messages if `mvbox_move` is off: Inbox or Sent? (currently it's inbox)